tests/test_port.py

```diff
@@ -9,132 +9,56 @@
-
-    dist, nn = clf.kneighbors()
-
-    # assert_array_equal(nn, moscow_raw.ref_neighbors)
-    assert_array_almost_equal(dist, moscow_raw.ref_distances, decimal=3)
-
-    dist, nn = clf.kneighbors(X_test)
-
-    # assert_array_equal(nn, moscow_raw.trg_neighbors)
-    assert_array_almost_equal(dist, moscow_raw.trg_distances, decimal=3)
-
-    prd = clf.predict(X_test)
-    assert_array_almost_equal(prd, moscow_raw.trg_predicted_unweighted, decimal=3)
-
-    clf = RawKNNRegressor(n_neighbors=5, weights=yaimpute_weights).fit(X_train, y_train)
-    prd = clf.predict(X_test)
-    assert_array_almost_equal(prd, moscow_raw.trg_predicted_weighted, decimal=3)
-
-
-def test_moscow_euclidean(moscow_euclidean):
-    X_train, X_test, y_train, _ = train_test_split(
-        moscow_euclidean.X, moscow_euclidean.y, train_size=0.8, shuffle=False
-    )
-    clf = EuclideanKNNRegressor(n_neighbors=5).fit(X_train, y_train)
-
-    dist, nn = clf.kneighbors()
-
-    # assert_array_equal(nn, moscow_euclidean.ref_neighbors)
-    assert_array_almost_equal(dist, moscow_euclidean.ref_distances, decimal=3)
-
-    dist, nn = clf.kneighbors(X_test)
-
-    # assert_array_equal(nn, moscow_euclidean.trg_neighbors)
-    assert_array_almost_equal(dist, moscow_euclidean.trg_distances, decimal=3)
-
-    prd = clf.predict(X_test)
-    assert_array_almost_equal(prd, moscow_euclidean.trg_predicted_unweighted, decimal=3)
-
-    clf = EuclideanKNNRegressor(n_neighbors=5, weights=yaimpute_weights).fit(
-        X_train, y_train
-    )
-    prd = clf.predict(X_test)
-    assert_array_almost_equal(prd, moscow_euclidean.trg_predicted_weighted, decimal=3)
+ESTIMATOR_RESULTS = {
+    "raw": RawKNNRegressor,
+    "euclidean": EuclideanKNNRegressor,
+    "mahalanobis": MahalanobisKNNRegressor,
+    "gnn": GNNRegressor,
+    "msn": MSNRegressor,
+}
 
 
-def test_moscow_mahalanobis(moscow_mahalanobis):
-    X_train, X_test, y_train, _ = train_test_split(
-        moscow_mahalanobis.X, moscow_mahalanobis.y, train_size=0.8, shuffle=False
-    )
-    clf = MahalanobisKNNRegressor(n_neighbors=5).fit(X_train, y_train)
+def yaimpute_weights(d):
+    return 1.0 / (1.0 + d)
 
-    dist, nn = clf.kneighbors()
 
-    # assert_array_equal(nn, moscow_mahalanobis.ref_neighbors)
-    assert_array_almost_equal(dist, moscow_mahalanobis.ref_distances, decimal=3)
+@pytest.mark.parametrize(
+    "result", ESTIMATOR_RESULTS.items(), ids=ESTIMATOR_RESULTS.keys()
+)
+def test_kneighbors(result):
+    """Test that the ported estimators identify the correct neighbors and distances."""
+    method, estimator = result
+    dataset = load_moscow_stjoes_results(method=method)
 
-    dist, nn = clf.kneighbors(X_test)
+    est = estimator(n_neighbors=5).fit(dataset.X_train, dataset.y_train)
 
-    # assert_array_equal(nn, moscow_mahalanobis.trg_neighbors)
-    assert_array_almost_equal(dist, moscow_mahalanobis.trg_distances, decimal=3)
+    dist, nn = est.kneighbors()
+    # assert_array_equal(nn, dataset.ref_neighbors)
```

Comment:
I can't remember where we are on this, but are we waiting to test the IDs once we circle back to #37?  I think that's right, so we want to leave this commented for this PR, right?

Reply:
Yes, I think as soon as we merge these changes into #37, we'll be able to finally implement those neighbor ID checks for real.